config/v1/types_ingress.go

```diff
@@ -41,13 +44,86 @@ type IngressSpec struct {
 	// certificate.
 	// +optional
 	AppsDomain string `json:"appsDomain,omitempty"`
+
+	// ComponentRoutes is a list of routes configured by a cluster-admin.
+	// The namespace and name of each componentRoute should match an existing Route.
+	//
+	// To determine the set of configurable Routes, look at namespace and name of entries in the
+	// .status.componentRoutes, where participating operators place current route status.
+	ComponentRoutes []ComponentRouteSpec `json:"componentRoutes,omitempty"`
 }
 
 type IngressStatus struct {
+	// ComponentRoutes is where participating operators place the current route status for routes which the cluster-admin
+	// can customize hostnames and serving certificates.
+	//
+	// An operator that creates entries in this slice should clean them up during removal.
+	// An operator must also handle the case of deleted status without churn.
+	ComponentRoutes []ComponentRouteStatus `json:"componentRoutes,omitempty"`
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// ComponentRouteSpec allows for configuration of a route's hostname and serving certificate.
+type ComponentRouteSpec struct {
+	// Namespace is the namespace of the route to customize.  It must be a real namespace.  Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	// +required
+	Namespace string `json:"namespace"`
```

Comment:
kubebuilder required tag too please. Kubebuilder does not understand these generic ones.

Reply:
There is a pattern in kube for namespaces. Now you could pass an empty string or some garbage

Reply:
Good point! I purposely attempted to create a namespace with an invalid name and got the regex:
```
The Namespace "B&D-NAME" is invalid: metadata.name: Invalid value: "B&D-NAME": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
````

Should be good to go.